packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -111,7 +125,9 @@ const loadTSConfigFile = async (
     configObject = await configObject();
   }
 
-  registerer.enabled(false);
+  if (tsNode) {
+    registerer.enabled(false);
+  }
```

Comment:
The registerer needs to be disabled before this function returns because otherwise it will continue to run on related files which will mess with Jest's transformation abilities and cause a bunch of weird bugs (e.g. Jest can end up getting js versions of ts files, which'll not get mapped back to their sources when showing failure messages & stack traces).

Sadly it looks like `esbuild-register` doesn't support being enabled and disabled, so that will a block - however it shouldn't be hard to implement and if it's done with the same API as `ts-node` we should be able to leave these `enabled` calls as is.

Reply:
Should I open an issue on `esbuild-register`?